src/main/java/com/monumental/util/async/AsyncJob.java

```diff
@@ -0,0 +1,45 @@
+package com.monumental.util.async;
+
+import java.util.concurrent.CompletableFuture;
+
+public class AsyncJob {
```

Comment:
The `AsyncJob` class wraps the `CompletableFuture` class from the Java Concurrency API. `AsyncJob` gives us the ability to give jobs an Id, store a `CompletableFuture`, and have a progress value (double) associated with the `CompletableFuture